hazelcast/proxy/map.py

```diff
@@ -725,6 +725,21 @@ def set(self, key, value, ttl=-1):
         value_data = self._to_data(value)
         return self._set_internal(key_data, value_data, ttl)
 
+    def set_ttl(self, key, ttl):
+        """
+        Updates the TTL (time to live) value of the entry specified by the given key with a new TTL value. New TTL 
+        value is valid starting from the time this operation is invoked, not since the time the entry was created.
+        If the entry does not exist or is already expired, this call has no effect.
+
+        :param key: (object), the key of the map entry.
+        :param ttl: (int), maximum time for this entry to stay in the map (0 means infinite,
+            negative means map config default)
+        """
+        check_not_none(key, "key can't be None")
+        check_not_none(ttl, "ttl can't be None")
+        key_data = self._to_data(key)
```

Comment:
I see that in the Java client, near cache invalidates the key once you call the `set_ttl` on it. We should also do that here.

Basically,  you should move the `return self._encode_invoke_on_key(map_set_ttl_codec, key_data, key=key_data, ttl=to_millis(ttl))` part to a method called `_set_internal` and in the near cache implementation, invalidate the key. Something like this https://github.com/hazelcast/hazelcast-python-client/blob/master/hazelcast/proxy/map.py#L1002